### PR TITLE
sqlite3_value_text: null-terminated string.

### PR DESCRIPTION
**Pull request by tiagokepe**
Hey @hannesmuehleisen 

This PR changes the _sqlite3_value_text_ to return a null-terminated string.

### CODE REVIEW

File: tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp [hunk 1]
```diff
@@ -1588,9 +1587,17 @@ const unsigned char *sqlite3_value_text(sqlite3_value *pVal) {
 	}
 
 	if (pVal->type == SQLiteTypeValue::TEXT || pVal->type == SQLiteTypeValue::BLOB) {
-		// We need don't need to append \0 at the end of string
-		// To know the number of chars or bytes, call function sqlite3_value_bytes()
-		return (const unsigned char *)pVal->str_t.GetDataUnsafe();
+		auto length = pVal->str_t.GetSize();
+		// new string including space for the null-terminated char ('\0')
+		pVal->zMalloc = (char *)malloc(sizeof(char) * length + 1);
```
Comment (Mytherin) [inline tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp:1592]: What if `sqlite3_value_text` gets called multiple times on the same value? Wouldn't this leak? Perhaps you could check if zMalloc already exists, and if it exists just return the value in there?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good. Some comments:

File: tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp [hunk 2]
```diff
@@ -1587,6 +1587,10 @@ const unsigned char *sqlite3_value_text(sqlite3_value *pVal) {
 	}
 
 	if (pVal->type == SQLiteTypeValue::TEXT || pVal->type == SQLiteTypeValue::BLOB) {
+		// check if the string has already been allocated
+		if (pVal->szMalloc > 0) {
+			return (const unsigned char *)pVal->zMalloc;
```
Comment (Mytherin) [inline tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp:1592]: The same problem would still occur if we called `sqlite3_value_text` twice on a `SQLiteTypeValue::INTEGER` or `SQLiteTypeValue::FLOAT`, right?  Could you add code that fixes it in that case as well + add a test case for that? Thanks!

File: tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp (hunk 1 above)
Comment (tiagokepe) [reply tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp:1596]: Yeah, this cloud happen. I changed the code and added a test.

File: tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp (hunk 2 above)
Comment (tiagokepe) [reply tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp]: Good catch, I will add a new test case for that. Thanks!

**Review by Mytherin — APPROVED** [review]

File: tools/sqlite3_api_wrapper/test/include/udf_scalar_functions.hpp [hunk 3]
```diff
@@ -196,3 +210,23 @@ static void sum_overload_function(sqlite3_context *context, int argc, sqlite3_va
...
+	       sqlite3_value_type(argv[0]) == SQLITE_TEXT);
+	char *str = (char *)sqlite3_value_text(argv[0]);
+	assert(sqlite3_value_type(argv[0]) == SQLITE_TEXT);
+	auto len = strlen(str);
+	size_t len2;
+	char *str2;
+	// calling sqlite3_value_text multiple times,  i.e., 10x
+	for (size_t i = 0; i < 10; ++i) {
+		str2 = (char *)sqlite3_value_text(argv[0]);
+		assert(str == str2);
```
Comment (Mytherin) [inline tools/sqlite3_api_wrapper/test/include/udf_scalar_functions.hpp:226]: Are these functions only used in tests? If so, the asserts should be converted into `REQUIRE`.  I think this is also where the unused variable warning comes from - `assert` is removed in release mode, and after the asserts are removed `len2` is unused. `REQUIRE` will not be compiled out.

**Pull request merged**